ui/appui-react/src/appui-react/content/ContentGroup.tsx

```diff
@@ -44,7 +45,7 @@ export interface ContentGroupProps {
  * @public
  */
 export abstract class ContentGroupProvider {
-  abstract provideContentGroup(props: FrontstageProps): Promise<ContentGroup>;
+  abstract provideContentGroup(props: FrontstageProps | FrontstageConfig): Promise<ContentGroup>; // eslint-disable-line deprecation/deprecation
```

Comment:
@raplemie this might hurt at runtime if i.e. a custom `ContentGroupProvider` is updated to use a configuration, but the `FrontstageProvider` is not (or vice versa). The TS is happy to narrow down the types though..

Reply:
Yes, we cant have that like it given that it is possible to mix and match providers in frontstages that we dont own.

I think we'd need a different method name to get away with clashes, but that will still require us to be able to convert from a Props to a Config so we can pass it whatever it expects, which is not a simple thing.

```suggestion
  /**
   * Return the contentGroup based on the `FrontstageProps` received.
   * @deprecated Implement using `contentGroup(FrontstageConfig)` instead, no need to provide this method if the other is provided. At least one should be provided in order for the provider to do anything.
   */
  abstract provideContentGroup?(props: FrontstageProps): Promise<ContentGroup>; // eslint-disable-line deprecation/deprecation
  
  /**
   * Return the contentGroup based on the `FrontstageConfig`.
   * @beta This method will be required in upcoming version, this method will be prioritized if it exists over `provideContentGroup`. Only one of these 2 methods will be called and at least one need to be implemented in order for the provider to do anything.)
   */
  abstract contentGroup?(config: FrontstageConfig): Promise<ContentGroup>;
```

Reply:
Seems all good